### DISCUSSION TITLE
Verify Normals and Colors

### DISCUSSION THREAD

**Discussion by Markemp** [Markemp/Cryengine-Converter#94] [Q&A] [answered]
Looking for some people to test out the updated VertsUvs datastream processing, especially Max users who can view Normals straight from the .dae file.  I've attached a few files and I'm curious if the Normals processing (sometimes from a Normals datastream, sometimes from Tangents, sometimes from QTangents) is right.  I think the colors thing is good at least, but worth taking a look at as well if you have the time.

The `assassin_body.dae` file is from Hunt, and the normals are calculated from the QTangents datastream.

The `brfl_fps_behr_p4ar_body` is from SC, and the normals are from the Normals datastream (it also has Tangents).

The `AEGS_Avenger` is also from SC, with the same derivation for normals.  This file has some weird additional vertex info (both a VertsUv and Vertices datastream for each node), with the Vertices datastream having 4 bytes.  Based on traditional positioning of datastreams for VertsUVs, it would be a Normals chunk.

[ResourceFiles.zip](https://github.com/Markemp/Cryengine-Converter/files/6920615/ResourceFiles.zip)

**Comment by Baconator650** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/78010776/127951676-4b1b8a95-e23d-43f5-9746-f363e2625646.png)


![image](https://user-images.githubusercontent.com/78010776/127951685-3ffc09a5-5f7b-40d0-a7c3-a46158a9c4e4.png)


![image](https://user-images.githubusercontent.com/78010776/127951694-278afd35-7693-4577-a951-3a6e2ec5ed57.png)

Are these what you are requesting? I also applied to "weighted normal" to brfl_fps_behr_p4ar_body and didn't noticed any changes.

**Reply by Markemp** [reply]
Blender tends to recalculate the normals when the Collada file is loaded.  Apparently 3DS Max will use the values in the file for the normals, and it definitely shows some weird angles that are clearly not right.

I'm not sure what weighted normal does.  What is it weighted with?

edit:  I'm assuming that's Blender?  Unless Blender and 3DS Max are looking a lot more similar these days... :)

**Reply by Baconator650** [reply]
This is 3DS Max with "editable normals" modifier so that normals are visible.

**Reply by Markemp** [reply]
I've spent so much time failing to solve the normals issue I just get skeptical when I think it may actually be fully solved. :)

Thank you for testing these files out!

**Reply by Baconator650** [reply]
No problem. I can also pull up .cga files through Dymek's Modded Cry Engine and use the FBX exporter there to export the models for comparison if you would like.

**Reply by vmxeo** [reply]
> I'm not sure what weighted normal does. What is it weighted with?

Weighted with face size. great for hard surface modelling where you want to smooth the beveled edges but keep the larger surfaces flat. 
for reference: http://wiki.polycount.com/wiki/Face_weighted_normals
